tempesta_fw/str.h

```diff
@@ -209,20 +209,23 @@ size_t tfw_ultoa(unsigned long ai, char *buf, unsigned int len);
  * @flags	- 3 most significant bytes for number of chunks of compound
  * 		  string and the least significant byte for flags;
  */
-typedef struct {
-	void		*ptr;
+typedef struct tfwstr_t {
+	union {
+		char *data;
+		struct tfwstr_t *chunks;
+	} ptr;
```

Comment:
Totally agree with @i-rinat , need to make the union anonymous to address `data` and `chunks` directly.